src/main/java/edu/utdallas/davisbase/compiler/Compiler.java

```diff
@@ -366,6 +373,38 @@ public boolean checkIsNotNull(List<String> columnSpecs) {
     return false;
   }
 
+  /**
+   * @param columnSpecs column constraints for give column
+   * @return whether column is UNIQUE
+   */
+  public boolean checkIsUnique(List<String> columnSpecs) {
+    if (null != columnSpecs) {
+      if(columnSpecs.contains("UNIQUE")){
+        return true;
+      }
+    }
+    return false;
+  }
+
+  /**
+   * @param columnSpecs column constraints for give column
+   * @return whether column is PRIMARY KEY
+   */
+  public boolean checkIsPrimaryKey(List<String> columnSpecs, Index index, String columnName) {
+    if (null != columnSpecs) {
+      for (int lcv = 0; lcv < columnSpecs.size() - 1; lcv++) {
+        if (columnSpecs.get(lcv).equalsIgnoreCase("PRIMARY")
+          && columnSpecs.get(lcv + 1).equalsIgnoreCase("KEY")) {
+          return true;
+        }
+      }
+    }
+    if(null!= index && index.getColumnsNames().get(0).equalsIgnoreCase(columnName)){
```

Comment:
Do we need to validate that the `Index` given by JSqlParser is a `PRIMARY KEY` index? Or does JSqlParser only ever permit one index (which by default is primary) in a `CREATE TABLE` statement anyway?

Reply:
The parser validates that the user is only able to create a table with a single index for the pk